pkg/asset/manifests/clusterapi/cluster.go

```diff
@@ -0,0 +1,229 @@
+func (c *Cluster) Generate(dependencies asset.Parents) error {
+	installConfig := &installconfig.InstallConfig{}
+	clusterID := &installconfig.ClusterID{}
+	openshiftInstall := &openshiftinstall.Config{}
+	featureGate := &manifests.FeatureGate{}
+	bootstrapIgnAsset := &bootstrap.Bootstrap{}
+	masterIgnAsset := &machine.Master{}
+	rhcosImage := new(rhcos.Image)
+	dependencies.Get(installConfig, clusterID, openshiftInstall, bootstrapIgnAsset, masterIgnAsset, featureGate, rhcosImage)
+
+	// If the feature gate is not enabled, do not generate any manifests.
+	if !capiutils.IsEnabled(installConfig) {
+		return nil
+	}
+
+	c.FileList = []*asset.RuntimeFile{}
+
+	namespace := &corev1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: capiutils.Namespace,
+		},
+	}
+	c.FileList = append(c.FileList, &asset.RuntimeFile{Object: namespace, File: asset.File{Filename: "00_capi-namespace.yaml"}})
+
+	cluster := &clusterv1.Cluster{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      clusterID.InfraID,
+			Namespace: capiutils.Namespace,
+		},
+		Spec: clusterv1.ClusterSpec{},
+	}
+	c.FileList = append(c.FileList, &asset.RuntimeFile{Object: cluster, File: asset.File{Filename: "01_capi-cluster.yaml"}})
+
+	// Gather the ignition files, and store them in a secret.
+	{
+		masterIgn := string(masterIgnAsset.Files()[0].Data)
+		bootstrapIgn, err := injectInstallInfo(bootstrapIgnAsset.Files()[0].Data)
+		if err != nil {
+			return errors.Wrap(err, "unable to inject installation info")
+		}
+		c.FileList = append(c.FileList,
+			&asset.RuntimeFile{
+				File: asset.File{Filename: "01_ignition-secret-master.yaml"},
+				Object: &corev1.Secret{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      fmt.Sprintf("%s-%s", clusterID.InfraID, "master"),
+						Namespace: capiutils.Namespace,
+						Labels: map[string]string{
+							"cluster.x-k8s.io/cluster-name": clusterID.InfraID,
+						},
+					},
+					Data: map[string][]byte{
+						"format": []byte("ignition"),
+						"value":  []byte(masterIgn),
+					},
+				},
+			},
+			&asset.RuntimeFile{
+				File: asset.File{Filename: "01_ignition-secret-bootstrap.yaml"},
+				Object: &corev1.Secret{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      fmt.Sprintf("%s-%s", clusterID.InfraID, "bootstrap"),
+						Namespace: capiutils.Namespace,
+						Labels: map[string]string{
+							"cluster.x-k8s.io/cluster-name": clusterID.InfraID,
+						},
+					},
+					Data: map[string][]byte{
+						"format": []byte("ignition"),
+						"value":  []byte(bootstrapIgn),
+					},
+				},
+			},
+		)
+	}
+
+	var out *capiutils.GenerateClusterAssetsOutput
+	switch platform := installConfig.Config.Platform.Name(); platform {
+	case "aws":
+		// Move this somewhere else.
+		// if err := aws.PutIAMRoles(clusterID.InfraID, installConfig); err != nil {
+		// 	return errors.Wrap(err, "failed to create IAM roles")
+		// }
+		var err error
+		out, err = aws.GenerateClusterAssets(installConfig, clusterID)
```

Comment:
We are only using this output to grab the infrastructure ref, but we should also gather all the manifests and serialize them